functorch/csrc/TensorWrapper.h

```diff
@@ -38,6 +38,7 @@ struct FUNCTORCH_API TensorWrapper : public c10::TensorImpl {
       Tensor value,
       int64_t level,
       std::shared_ptr<bool> is_alive,
+      bool generated = false,  // if false underlying was passed as input to this level, otherwise wrapper was generated by interpreter
```

Comment:
> However, even unwrapped tensors need to be wrapped by the DyanmicLayerFrontMode key. So, this adds a "generated" value to the TensorWrapper that is true only if the tensor was created by the DyanmicLayerFrontMode because the original tensor was unwrapped. Then, we skip wrapping all the outputs that are aliased to inputs that were either unwrapped or generated

I don't understand why the `generated` is necessary: why do unwrapped tensors need to be wrapped by the DynamicLayerFrontMode key?

Reply:
Is it necessary for the case where some inputs are wrapped and some inputs aren't wrapped?

Reply:
Answering in reverse order
> why do unwrapped tensors need to be wrapped by the DynamicLayerFrontMode key? Is it necessary for the case where some inputs are wrapped and some inputs aren't wrapped?

Yes it should only come up where some inputs are wrapped and others aren't. The unwrapped ones are wrapped right before they run through autograd so that the AutogradMeta information gets to go on the right tensor. Also if the unwrapped tensor has any autograd metadata already on it, autograd gets mad if it sees the unwrapped version since autograd meta and tensors are 1-1

> I don't understand why the generated is necessary

This might have already been clear but just for extra clarity, once the tensors are wrapped, they all look the same so I was thinking that we have to pass the information about which tensors were originally wrapped from the transform to the send to next interpreter functions. Since the interpreters don't currently have state that's not like the constant state of the transform, it felt easier to attach it to the tensors themselves